### ISSUE TITLE
Feature Request: "Always Show Pictures for address/domain"

### ISSUE THREAD

**Issue by supertin** [k9mail/k-9#1035]
As well as the "always show for contacts" option for pictures, can we also have some sort of whitelisting to work alongside?

I get a number of genuine advertising type mailing lists, most of which I would like to see the pictures for. But to show pictures for these at the moment, I either have to press the button every time, add them to my contact list, or allow pictures always.
Adding them to my contacts would be OK if the contacts were only for use in K9,  but it appears to be wanting to add a global phone contact, which is silly for this purpose.

**Comment by younes778** [comment]
Class UI/messageView/MessageTopView
Methode : shouldAutomaticallyLoadPictures
 return showPicturesSetting == ShowPictures.ALWAYS || shouldShowPicturesFromSender(showPicturesSetting, message) || isTrustedSender(message.getFrom());
then add methode isTrustedSender and return true the send is in the trusted list.

**Comment by philipwhiuk** [comment]
Please provide patches as pull requests not in issue comments.

**Comment by younes778** [comment]
Sorry, just proposing a way to resolve the issue. The general solution may be more complete.

**Comment by markreg** [comment]
I second this enhancement and think that this is the only thing really missing from K9.

I don't want to add every potentially whitelisted address/domain to my phone's contacts but I also don't want to allow images from all addresses/domains coming into a particular email account.

A global or per-account whitelist for all the address/domains for which pictures are shown automatically would really cut down on having to press "Show Images" about 70% of the time in one of my email accounts.

**Comment by bokkiedog** [comment]
I would like this feature as well.

**Comment by ledeniz** [comment]
I agree. I use K9 for several years and this missing feature is the only reason I look at other clients right now.

**Comment by DasWookie** [comment]
+1! I am CONSTANTLY having to hit the "Show Pictures" for messages who are being sent from accounts or domains who I do not have in my contacts, either because they are send only accounts or are sent from multiple senders from a domain I trust, but don't have EVERY sender in my contacts. This would be a MUCH improved user experience!

**Comment by DasWookie** [comment]
Bump: can't this please get assigned to someone to implement? This has been requested for a LONG time!

**Comment by beneficadoramocaba** [comment]
I was about to propose this feature, so I was quite happy to see it had just been assigned... Though I am guessing that is probably a joke? :smiling_face_with_tear:

**Comment by DasWookie** [comment]
Not a very good joke either. This feature is BADLY needed.

**Comment by cketti** [comment]
We don't prioritize features by listening to who shouts the loudest. But it's an open source project. So there's at least the possibility of implementing the feature yourself (or paying someone to write the code for you).

Adding a comment requesting that something should happen does nothing to get the feature implemented any faster. It probably has the opposite effect, because it takes time to read such comments and maybe respond to them. That's also the reason why such comments will be hidden. The fact that someone needs this feature badly, is not relevant when a developer finally gets around to implementing the feature. I don't want anyone to have to waste time reading those comments (again).
